### PR TITLE
Replace logger.warnings with warnings.warn

### LINKED ISSUE #1328: [FEAT] Use warnings in a single way
Right now we have 2 types of warnings, using `logger` and using `warnings`. We should decide which one is better to use and then update in all places to use that one

### PR DESCRIPTION
**Pull request by matanper**
close #1328

### CODE REVIEW

File: tests/base/dataset_test.py [hunk 1]
```diff
@@ -1041,17 +1042,16 @@ def random_classification_dataframe(n_samples=100, n_features=5) -> pd.DataFrame
 
 def test_cat_features_warning(iris, caplog):
```
Comment (ItayGabbay) [inline tests/base/dataset_test.py:1043]: you don't need the caplog anymore

File: deepchecks/utils/distribution/drift.py [hunk 2]
```diff
@@ -294,7 +294,7 @@ def calc_drift_and_plot(train_column: pd.Series,
             'largest_difference': 'largest difference between categories'
         }
         train_data_percents = dist_traces[0].y.sum()
```
Comment (ItayGabbay) [inline deepchecks/utils/distribution/drift.py:296]: ?

**Review by ItayGabbay — APPROVED** [review]

File: deepchecks/tabular/dataset.py [hunk 3]
```diff
@@ -12,6 +12,7 @@
 import logging
```
Comment (Nadav-Barak) [inline deepchecks/tabular/dataset.py:12]: what is logging used for?

**Comment by JKL98ISR** [comment]
remove all `logger = logging.getLogger('deepchecks')` and such (no import logging should be there)

**Pull request merged**